python/cudf/cudf/core/series.py

```diff
@@ -1510,6 +1510,94 @@ def fillna(
             value=value, method=method, axis=axis, inplace=inplace, limit=limit
         )
 
+    def between(self, left, right, inclusive="both") -> Series:
+        """
+        Return boolean Series equivalent to left <= series <= right.
+
+        This function returns a boolean vector containing `True` wherever the
+        corresponding Series element is between the boundary values `left` and
+        `right`. NA values are treated as `False`.
+
+        Parameters
+        ----------
+        left : scalar or list-like
+            Left boundary.
+        right : scalar or list-like
+            Right boundary.
+        inclusive : {"both", "neither", "left", "right"}
+            Include boundaries. Whether to set each bound as closed or open.
+
+        Returns
+        -------
+        Series
+            Series representing whether each element is between left and
+            right (inclusive).
+
+        See Also
+        --------
+        Series.gt : Greater than of series and other.
+        Series.lt : Less than of series and other.
+
+        Notes
+        -----
+        This function is equivalent to ``(left <= ser) & (ser <= right)``
+
+        Examples
+        --------
+        >>> import cudf
+        >>> s = cudf.Series([2, 0, 4, 8, None])
+
+        Boundary values are included by default:
+
+        >>> s.between(1, 4)
+        0     True
+        1    False
+        2     True
+        3    False
+        4     <NA>
+        dtype: bool
+
+        With `inclusive` set to ``"neither"`` boundary values are excluded:
+
+        >>> s.between(1, 4, inclusive="neither")
+        0     True
+        1    False
+        2    False
+        3    False
+        4     <NA>
+        dtype: bool
+
+        `left` and `right` can be any scalar value:
+
+        >>> s = cudf.Series(['Alice', 'Bob', 'Carol', 'Eve'])
+        >>> s.between('Anna', 'Daniel')
+        0    False
+        1     True
+        2     True
+        3    False
+        dtype: bool
+        """
+
+        if inclusive == "both":
+            lmask = self >= left
```

Comment:
Do we need to worry about any weirdness / asymmetry in how binary ops are called? Is it possible for any dtypes or input objects to have a result where `self >= left` is not equal to (or a different type than) `left <= self`? If so, is that case potentially problematic here? (This implementation matches pandas but it claims to match ``(left <= ser) & (ser <= right)`` which is not exactly what is written here.)

I think the answer is, "No, this is not a concern. You're thinking too hard on a Friday night and the current implementation is fine." I just wanted to raise that question in case someone can think of an edge case to this behavior.

Reply:
I tried this out on other dtypes, and we error at binops level for list & struct dtypes with a meaningful not implemented error message. So the result types is always the same for all `>`, `<`, `=` combinations. So this shouldn't be an issue.

Reply:
We'll support binops on nested types at some point soon (#10186 defines some preliminary steps, and #9452 is evolving in that direction as well). The question about input object types (list, pandas Series, etc.) is nicely solved by your use of `as_column`. This can be resolved.